### DISCUSSION TITLE
datasets.zip error - BGYSC3908E Invalid dataset name

### DISCUSSION THREAD

**Discussion by gngrossi** [IBM/zoau-samples#59] [Q&A] [answered]
For the first time, trying to use datasets.zip. Seeing BGYSC3908E Invalid dataset name provided: /tmp/gary/asmdb2
Also, a bit confused on the datasets.unzip API, for hlq, can I use a z/OS dataset name?
thanks

datasets.unzip
Description
Unzips a .dzp file.
Returns
Return code from z/OS.
Return type
int
Parameters
file (str) – HFS path or data set containing backup.
hlq (str) – Single segment HLQ to unpack to.

--

>>> datasets.zip("@02858.T.ASMDB2", "/tmp/gary/asmdb2", debug=True)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/hewitt/zopentools/zoautil/lib/zoautil_py/datasets.py", line 930, in zip
    raise ZOAUException("%s,RC=%s" % (response.stderr_response, response.rc))
zoautil_py.exceptions.ZOAUException: + expr 2 - 1
+ 1<TMP> /u/@02858/tmp/shFFBhDbdfA
+ shift 1
+ [[ 2 -lt 2 ]]
+ dest=@02858.T.ASMDB2
+ shift 1
+ [[ \0 -eq 0 ]]
+ parse_dataset_str include /tmp/gary/asmdb2
+ dataset_array=include
+ shift 1
+ dataset_names=/tmp/gary/asmdb2
+ datasets_patterns=
+ dataset_list=
+ 1<TMP> /u/@02858/tmp/shAcbDCBafA
+ echo /tmp/gary/asmdb2
+ grep -q ,
+ [[ -eq 0 ]]
+ + 1<TMP> /u/@02858/tmp/shchECgcHgA
+ echo /tmp/gary/asmdb2
+ tr ,
dataset_list=/tmp/gary/asmdb2
+ check_dataset_pattern /tmp/gary/asmdb2
+ dataset_name=/tmp/gary/asmdb2
+ echo /tmp/gary/asmdb2
+ grep -Eq ^(([A-Za-z$#@\%\?]{1}[A-Za-z0-9$#@\-\%\?]{0,7})([.]{1})){1,21}[A-Za-z$#@\*\%\?]{1}[A-Za-z0-9$#@\-\*\%\?]{0,7}$
+ retVal=1
+ [[ 1 -ne 0 ]]
+ mvscmdmsghelper dzip 8 E /tmp/gary/asmdb2
BGYSC3908E Invalid dataset name provided: /tmp/gary/asmdb2
+ exit 4
+ cleanup
+ rc=4
+ [ ! -z  ]
+ [ ! -z  ]
+ [ ! -z @02858.T.ASMDB2 -a 0 -eq 0 ]
+ [ 0 -eq 0 -a 4 -gt 0 ]
+ rm @02858.T.ASMDB2
+ 1> /dev/null 2<& 1
,RC=4

**Comment by EnsonoStevenPerva** [comment, marked as answer]
I believe your call to `Datasets.zip(...)` has the arguments swapped. I was able to create a zipped up PDSE with this code:
```
Python 3.10.0 (heads/pyz_dev-3.10:052d330082, Nov 18 2021, 12:17:59) [Clang 4.0.1 (tags/RELEASE_401/final)] on zos
Type "help", "copyright", "credits" or "license" for more information.
>>> from zoautil_py import datasets
>>> resp = datasets.zip('/u/ibmuser/test.dzp','IBMUSER.JCLLIB')
>>> print(resp)
0
>>>
```

As for `datasets.unzip(...)` you can only change the HLQ, so the method is requesting an override for the HLQ when you call it. One of the best ways to see what the Python API is teasing at is to take a look at the `man` page for the command the API is calling under the hood (in this case dunzip). Assuming you got your `man` pages sorted, do a `man dunzip`. If you don't, let me know and I'll paste the contents of it here for you.

**Reply by gngrossi** [reply]
Thanks for the insight...much appreciated. As you indicated, I misinterpreted the parameters (looking at target being the .dzp file).

A doc snippet from the web. https://www.ibm.com/docs/en/zoau/1.2.x?topic=modules-datasets

zoautil_py.datasets.zip(file, target, *args, **kwargs)
Description
Zip datasets into an HFS file.
Returns
Return code from z/OS.
Return type
int
Parameters
file (str) – Fully qualified HFS file path or data set.
target (str) – Dataset name or dataset pattern.

--

zip result:

datasets.zip("/tmp/gary/asmdb2.dzp", "@02858.T.ASMDB2")
0

--

For unzip, was able to read the man page for dunzip and able to better understand it.
Luckily, we have an HLQ we can use for the way it works.

DUNZIP(1)                  ZOA Utilities Command Syntax                DUNZIP(1)
NAME
        dunzip: unzip a dzip binary file to specified HLQ
DESCRIPTION
    Unzips a file created by dzip, with the option of changing the first
    segment of the HLQ. (e.g USER.TEST.FILE -> MY.TEST.FILE)
    Specifying more than the first HLQ segment (e.g MY.HLQ) will result
    in an error.

--

unzip result:
datasets.unzip("/tmp/gary/asmdb2.dzp", "TEMP")
0

thanks again
